packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -333,7 +333,75 @@ class ShaderMask extends SingleChildRenderObjectWidget {
   }
 }
 
-/// A widget that applies a filter to the existing painted content and then paints [child].
+/// A widget that applies a filter to the existing painted content and then
+/// paints [child].
+///
+/// The filter will only be applied to the area of the background in which the
+/// [child] (or one of its descendants) is actually going to paint in regardless
+/// of the actual size of [child].
+///
+/// In more technical terms, [BackdropFilter] will only apply the filter to the
+/// area of the background that is covered by the [Scene] defined by [child]
+/// and its descendants.
+///
+/// {@tool sample}
+/// Even though the [BackdropFilter] is wrapping the [Container] below, the
+/// background will only be blurred in the area defined by the bounding box
+/// of the [Text] because that's the only area any descendant of the
+/// [BackdropFilter] is painting in.
+///
+/// ```dart
+/// Stack(
+///   fit: StackFit.expand,
+///   children: <Widget>[
+///     Text('0' * 10000),
+///     Center(
+///       child: BackdropFilter(
+///         filter: ui.ImageFilter.blur(
+///           sigmaX: 5.0,
+///           sigmaY: 5.0,
+///         ),
+///         child: Container(
+///           alignment: Alignment.center,
+///           width: 200.0,
+///           height: 200.0,
+///           child: Text('Hello World'),
+///         ),
+///       ),
+///     ),
+///   ],
+/// )
+/// ```
+///
+/// To blur the entire area of the [Container], increase the paint area of
```

Comment:
Can we wrap/change the behavior of the BackdropFilter so users don't need to learn about this distinction? (or do they?)

Reply:
I would love to make this more intuitive. Do you have an idea how that can be achieved, though?

Reply:
Maybe there's a much smarter fix that can be done at the layer level, @liyuqian might know. Otherwise, out of convenience, we can just emulate the below behavior in the framework?

Reply:
https://github.com/flutter/flutter/pull/28174

Reply:
\o/

Reply:
After talking to @liyuqian we will do this in two steps: I am going to submit this PR with the updated documentation of how the BackdropFilter currently works. When @liyuqian finds some time, he will finish #28174 to change the behavior of this to something more reasonable.